exasol_integration_test_docker_environment/lib/api/common.py

```diff
@@ -86,42 +110,109 @@ def import_build_steps(flavor_path: Tuple[str, ...]):
     success = False
+    log_file_path = get_log_path(task.job_id)
     try:
-        with get_luigi_log_config(get_log_path()) as luigi_config:
-            no_scheduling_errors = luigi.build([task], workers=workers,
-                                               local_scheduler=True,
-                                               logging_conf_file=f'{luigi_config}')
+        no_scheduling_errors = \
+            _run_task_with_logging_config(
+                task, log_file_path, log_level, use_job_specific_log_file, workers)
         success = not task.failed_target.exists() and no_scheduling_errors
-        generate_graph_from_task_dependencies(task, task_dependencies_dot_file)
-
-        if success:
-            return task.get_result()
-        elif task.failed_target.exists():
-            logging.error(f"Task {task} failed. failed target exists.")
-            raise TaskRuntimeError(msg=f"Task {task} (or any of it's subtasks) failed.",
-                                   inner=list(task.collect_failures().keys()))
-        elif not no_scheduling_errors:
-            logging.error(f"Task {task} failed. : luigi reported a scheduling error.")
-            raise TaskRuntimeError(msg=f"Task {task} failed. reason: luigi reported a scheduling error.")
+        return _handle_task_result(no_scheduling_errors, success, task,
+                                   task_dependencies_dot_file)
     except BaseException as e:
         logging.error("Going to abort the task %s" % task)
         raise e
     finally:
+        if use_job_specific_log_file:
+            logging.info(
+                f"The detailed log of the integration-test-docker-environment can be found at: {log_file_path}")
         task.cleanup(success)
 
 
+def _run_task_with_logging_config(
+        task: DependencyLoggerBaseTask,
+        log_file_path: Path,
+        log_level: Optional[str],
+        use_job_specific_log_file: bool,
+        workers: int) -> bool:
+    with _configure_logging(log_file_path, log_level, use_job_specific_log_file) as run_kwargs:
+        no_scheduling_errors = luigi.build([task], workers=workers,
+                                           local_scheduler=True, **run_kwargs)
+        return no_scheduling_errors
+
+
+def _handle_task_result(no_scheduling_errors, success, task, task_dependencies_dot_file: Optional[str]) -> Any:
+    generate_graph_from_task_dependencies(task, task_dependencies_dot_file)
+    if success:
+        return task.get_result()
+    elif task.failed_target.exists():
+        logging.error(f"Task {task} failed. failed target exists.")
+        raise TaskRuntimeError(msg=f"Task {task} (or any of it's subtasks) failed.",
+                               inner=list(task.collect_failures().keys()))
+    elif not no_scheduling_errors:
+        logging.error(f"Task {task} failed. : luigi reported a scheduling error.")
+        raise TaskRuntimeError(msg=f"Task {task} failed. reason: luigi reported a scheduling error.")
```

Comment:
Is there no default case?

could this be an alternative:
```python
if task.failed_target.exists():
   ....
if not no_scheduling_errors:
   ....
   
return task.get_result()
 ```

Reply:
I only extracted that part, didn't think about it more. But, I think, this contains all possible cases,
```
        success = not task.failed_target.exists() and no_scheduling_errors
```

But, this construction is probably in general a it problematic.

Reply:
created a follow up task https://github.com/exasol/integration-test-docker-environment/issues/298